### PR TITLE
AJ-761 add spinning wheel for loading wds tables

### PR DESCRIPTION
**Pull request by calypsomatic**
See [AJ-761](https://broadworkbench.atlassian.net/browse/AJ-761)
For an azure workspace, if WDS is not ready, show a spinning wheel and a message that the database is being created.  Once it exists it will show the same "Select a data type from the navigation panel on the left" message as GCP workspaces.

image

[AJ-761]: https://broadworkbench.atlassian.net/browse/AJ-761?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/pages/workspaces/workspace/Data.js [hunk 1]
```diff
@@ -985,83 +985,84 @@ const WorkspaceData = _.flow(
...
-          })],
-          [workspaceDataTypes.wds, () => wdsDataTableProvider && proxyUrlLoaded && !_.isEmpty(wdsSchema) && h(WDSContent, {
-            key: refreshKey,
-            workspaceUUID: workspaceId,
-            workspace,
-            dataProvider: wdsDataTableProvider,
-            recordType: selectedData.entityType,
-            wdsSchema
-          })]
+        Utils.switchCase(selectedData?.type, [undefined, () => Utils.cond([wdsSchemaError && isAzureWorkspace, () => div({ style: { textAlign: 'center' } }, [icon('loadingSpinner'), ' The database that powers your data tables is being created. This may take a few minutes.'])],
```
Comment (jordansafer) [inline src/pages/workspaces/workspace/Data.js]: How come we check for `wdsSchemaError`? Maybe this is a dumb question, do we expect wdsSchemaError to be true before wds is loaded?

Comment (calypsomatic) [reply src/pages/workspaces/workspace/Data.js]: Hmm, my interpretation was that `wdsSchemaError` indicates that there's some problem with wds - hopefully just that the app is still deploying but potentially something else gone wrong.  It's initialized as `undefined` though which maybe is an issue?

**Review by davidangb — COMMENTED** [review]
Hm, so there are a bunch of conditions in which the text "The database … is being created" is misleading or incorrect.

If the user has deployed an app, but then stopped the app, they'll still see this text. If the user hasn't yet deployed an app, or if the auto-deploy failed, they'll still see this text. If the app is deployed and running but returning an error for some reason, they'll still see this text.

The text as written only seems applicable for the condition in which there is only one app, but that app is still in `PROVISIONING` status. We don't detect that condition right now, though `getWdsUrl` could potentially detect it.

I see two paths forward:
1. Work with PM to make the error text more generic, something like "The database that powers your data tables is unavailable. It may take a few minutes after initial workspace creation to be ready."
2. Detect the particular condition of a `PROVISIONING` app and display this message for it; display a different, more generic, message for other error conditions

See also AJ-781 - you could add another condition for `!proxyUrlLoaded` and display a different error message for that condition.

See also SU-270 - if that troubleshooter lands, I think it will help with identifying WDS problems and reduces the need to have nice specific error messages here.

**Comment by bethsheets** [comment]
@calypsomatic for the scope of this ticket, let's go with David's #1 example, "The database that powers your data tables is unavailable. It may take a few minutes after initial workspace creation to be ready."

I agree with following up to provide better messaging for different possible states of WDS after SU-270 is complete.

**Review by davidangb — APPROVED** [review]
:+1: with PM's blessing - this is an improvement over the current state.

**Review by jordansafer — APPROVED** [review]

**Pull request merged**